### PR TITLE
Add CSV delimiter to csv document handler methods

### PR DESCRIPTION
**Pull request by brunoocasali**
- Add `?string $delimiter` to `updateDocumentsCsv`, `addDocumentsCsv` and also `addDocumentsCsvInBatches` and `updateDocumentsCsvInBatches`.

### CODE REVIEW

File: tests/Endpoints/DocumentsTest.php [hunk 1]
```diff
@@ -280,6 +302,34 @@ public function testAddDocumentsCsvInBatches(): void
         $this->assertSame($total, $response->getTotal());
     }
 
+    public function testAddDocumentsCsvInBatchesWithDelimiter(): void
```
Comment (norkunas) [inline tests/Endpoints/DocumentsTest.php:305]: would it fail with empty delimiter? if yes, maybe better to document then as `non-empty-string|null` ?

File: tests/Endpoints/DocumentsTest.php [hunk 2]
```diff
@@ -610,6 +687,34 @@ public function testUpdateDocumentsCsvInBatches(): void
...
+
+        $index->expects($matcher)
+              ->method('updateDocumentsCsv')
+              ->willReturnCallback(function (string $param) use ($matcher): void {
+                  // withConsecutive has no replacement https://github.com/sebastianbergmann/phpunit/issues/4026
+                  // @phpstan-ignore-next-line
+                  match ($matcher->numberOfInvocations()) {
+                      1 => $this->assertEquals($param, ["id;title\n888221515;Young folks", null, ';']),
+                      2 => $this->assertEquals($param, ["id;title\n235115704;Mister Klein", null, ';']),
+                  };
```
Comment (norkunas) [inline tests/Endpoints/DocumentsTest.php:711]:
```suggestion
                      2 => $this->assertEquals($param, ["id;title\n235115704;Mister Klein", null, ';']),
                      default => self::fail(),
                  };
```

File: tests/Endpoints/DocumentsTest.php (hunk 1 above)
Comment (norkunas) [reply tests/Endpoints/DocumentsTest.php:305]: With this phpdoc users could know what it would fail even before calling the api

**Review by alallema — APPROVED** [review]
So great! ✨ Remind me that I didn't check the.`CsvInBatches` methods
LGTM!

File: tests/Endpoints/DocumentsTest.php (hunk 1 above)
Comment (brunoocasali) [reply tests/Endpoints/DocumentsTest.php:305]: Yes, thanks @alallema I'll also add your suggestion @norkunas :)

File: tests/Endpoints/DocumentsTest.php [hunk 3]
```diff
@@ -84,6 +84,28 @@ public function testAddDocumentsCsv(): void
         $this->assertCount(20, $response);
     }
 
+    public function testAddDocumentsCsvWithCustomSeparator(): void
+    {
+        $index = $this->client->index('documentCsvWithCustomSeparator');
+
+        $fileCsv = fopen('./tests/datasets/songs-custom-separator.csv', 'r');
+        $documentCsv = fread($fileCsv, filesize('./tests/datasets/songs-custom-separator.csv'));
+        fclose($fileCsv);
```
Comment (norkunas) [inline tests/Endpoints/DocumentsTest.php:93]: why not just `file_get_contents`?

Comment (brunoocasali) [reply tests/Endpoints/DocumentsTest.php:93]: Good idea! I'm not much aware of all the PHP features, so I'm glad you suggested that :)

**Pull request merged**